server/apns/apns_sender.go

```diff
@@ -43,10 +46,47 @@ func NewSenderUsingPusher(pusher Pusher, appTopic string) (connector.Sender, err
 func (s sender) Send(request connector.Request) (interface{}, error) {
 	deviceToken := request.Subscriber().Route().Get(deviceIDKey)
 	logger.WithField("deviceToken", deviceToken).Debug("Trying to push a message to APNS")
-	return s.client.Push(&apns2.Notification{
-		Priority:    apns2.PriorityHigh,
-		Topic:       s.appTopic,
-		DeviceToken: deviceToken,
-		Payload:     request.Message().Body,
-	})
+	push := func() (interface{}, error) {
+		return s.client.Push(&apns2.Notification{
+			Priority:    apns2.PriorityHigh,
+			Topic:       s.appTopic,
+			DeviceToken: deviceToken,
+			Payload:     request.Message().Body,
+		})
+	}
+	withRetry := &retryable{
+		Backoff: backoff.Backoff{
+			Min:    1 * time.Second,
+			Max:    10 * time.Second,
+			Factor: 2,
+			Jitter: true,
+		},
+		maxTries: 3,
+	}
+	return withRetry.execute(push)
```

Comment:
I am not really sure about sender recreation. I tested it locally and the client could recover after timeout or when the server was down. 
So I added a specific error, but so far the sender is not recreated. If this really should be done, the sender http client should probably also be closed to avoid resource leak.